apimatic_requests_client_adapter/requests_client.py

```diff
@@ -52,7 +52,8 @@ def __init__(self,
                     adapter.max_retries.allowed_methods = retry_methods
 
             self.timeout = http_client_instance.timeout
-            self.session = http_client_instance.session
+            if hasattr(http_client_instance, 'session'):
```

Comment:
Blocker: This same check should be added when overriding the session property in override_http_client_configuration flow.